src/common/server.ts

```diff
@@ -217,6 +232,15 @@ export async function startServer(
         }
       });
     }),
+
+    pythonExtension.environments.onDidChangeActiveEnvironmentPath(() => {
+      if (middleware.isDidChangeConfigurationRegistered()) {
+        logger.debug("Python interpreter changed, sending DidChangeConfigurationNotification");
+
+        // If the Python interpreter changes,
+        newLSClient.sendNotification(DidChangeConfigurationNotification.type, undefined);
+      }
```

Comment:
This is a bit overkill at the moment because we always restart the server when the interpreter changes (because we might need to use a different binary now). I kept the implementation because it took me a rather long time to figure out how to know if the server registered the capability but I can also remove it again. 

I think it would also be great if we can avoid restarting the LSP if the interpreter changes....

Reply:
Yeah, agreed. I think the only reason that we would use a different binary is if the ty binary used is installed in the virtual environment where the interpreter is present.

I don't know if we can avoid the restart completely but we could check if the interpreter is even going to be used to find the binary path. If it's not going to be used, then we can avoid restarting the server when the interpreter changes.

The checks that I'm referring to is the one before line 101 in the below code:

https://github.com/astral-sh/ty-vscode/blob/6e934d0f07b6750ff2face39aba08d5ca9c3757b/src/common/server.ts#L79-L136